common/persist.c

```diff
@@ -562,6 +562,13 @@ field_to_attribute (p11_persist *persist,
 		return false;
 	}
 
+	/* Ignore recursive attributes */
+	if (IS_ATTRIBUTE_ARRAY (&attr)) {
```

Comment:
Can we simply return `false` rather than proceeding with a NULL attribute?

Reply:
No, because the p11-kit-testable binary is using module-ep12 which at the beginning loads and at exit persists all objects into .p11-kit file. This is used in some tests to preserve the objects across multiple calls to p11-kit-testable. However as the mock_reset_objects function always creates the object with CKA_WRAP_TEMPLATE, the p11-kit-testable will always write this object at exit into .p11-kit file. If I were to make it fail here, it would break the tests because module setup will fail during reading of .p11-kit file. So I made it to just ignore the CKA_WRAP_TEMPLATE value.

Reply:
In that case, can we return `true` without adding it to the array?

Reply:
yes, we can.